File: python-app/models/slide/self.py
from models.image_slide.image_slide_kind import ImageSlideKind
from models.slide.slide_category import SlideCategory

from pandas import Timestamp
from abc import ABC, abstractmethod
from pptx.slide import Slide
from render.drawable_area import DrawableArea

class Slide(ABC):
    def __init__(self, 
            title: str,
            identifier: str, 
            creation_date: Timestamp,
            last_edit: Timestamp,
            preview: str | None,
            category: SlideCategory,
            ) -> None:
        self.title = title
        self.identifier = identifier
        self.creation_date = creation_date
        self.last_edit = last_edit
        self.preview = preview
        self.category = category

    def to_dict(self) -> dict[str, any]:
        return {
            "title": self.title,
            "identifier": self.identifier,
            "creation_date": self.creation_date.isoformat(),
            "last_edit": self.last_edit.isoformat(),
            "preview": self.preview,
            "category": self.category.value,
        }

    @abstractmethod
    def render(self, slide: Slide, drawable_area: DrawableArea):
        pass

File: lib/models/slide/self.dart
import 'package:ut_report_generator/models/slide_category.dart';

class Slide {
  String identifier;
  String title;
  DateTime creationDate;
  DateTime lastEdit;
  String preview;
  SlideCategory category;

  Slide({
    required this.identifier,
    required this.title,
    required this.creationDate,
    required this.lastEdit,
    required this.preview,
    required this.category,
  });

  Map<String, dynamic> toMap() {
    return {
      "title": title,
      "identifier": identifier,
      "creation_date": creationDate.toIso8601String(),
      "last_edit": lastEdit.toIso8601String(),
      "preview": preview,
      "category": category.name,
    };
  }
}
